File: javascript/bob/bob.js
function Bob() {
  'use strict';

  this.hey = function(input) {
    if (!input.trim()) {
        return "Fine. Be that way!";
    } else if (input === input.toUpperCase() && input !== input.toLowerCase()){
        return 'Whoa, chill out!';
    } else if (input.slice(-1) === '?'){
        return 'Sure.';
    } else {
        return 'Whatever.';
    }
  }
};

module.exports = Bob;

File: java/bob/src/main/java/Bob.java
public class Bob {
  public String hey(String s) {
    if (s.trim().isEmpty()) {
      return "Fine. Be that way!";
    } else if (s.equals(s.toUpperCase()) && !s.equals(s.toLowerCase())) {
      return "Whoa, chill out!";
    } else if (s.charAt(s.length() - 1) == '?') {
      return "Sure.";
    } else {
      return "Whatever.";
    }
  }
}
